src/outputs/history.cpp

```diff
@@ -102,24 +96,28 @@ void HistoryOutput::WriteOutputFile(Mesh *pm, ParameterInput *pin, SimTime *tm,
       md_base->Set(pm->block_list, pm);
     }
 
-    for (const auto &hist_var : hist_vars) {
-      auto result = hist_var.hst_fun(md_base.get());
-      results[hist_var.hst_op].push_back(result);
-      labels[hist_var.hst_op].push_back(hist_var.label);
+    // Check if the package has enrolled scalar history functions which are stored in the
+    // Params under the `hist_param_key` name.
+    if (params.hasKey(hist_param_key)) {
```

Comment:
Only potential change in behavior is that, even if no history vars are enrolled, we still update `md_base` it it was empty or if it had a different size than `pm->block_list`. I think this is fine because codes shouldn't be designed around not updating this variable during history output *only* if no history variables were requested.

I also now update `md_base` only once for all packages rather than inside the `for (const auto &pkg : packages)` loop.